addons/point_of_sale/models/pos_session.py

```diff
@@ -300,15 +305,485 @@ def _validate_session(self):
+                    exp_key = move.product_id.property_account_expense_id or move.product_id.categ_id.property_account_expense_categ_id
+                    out_key = move.product_id.categ_id.property_stock_account_output_categ_id
+                    amount = -sum(move.stock_valuation_layer_ids.mapped('value'))
+                    stock_expense[exp_key] = self._update_amounts(amount, move.picking_id.date, **stock_expense[exp_key])
+                    stock_output[out_key] = self._update_amounts(amount, move.picking_id.date, **stock_output[out_key])
+
+        ## SECTION: Create non-reconcilable move lines
+        # Create account.move.line records for
+        #   - sales
+        #   - taxes
+        #   - stock expense
+        #   - non-cash split receivables (not for automatic reconciliation)
+        #   - non-cash combine receivables (not for automatic reconciliation)
+        MoveLine = self.env['account.move.line'].with_context(check_move_validity=False)
+
+        tax_vals = [self._get_tax_vals(key, **amounts) for key, amounts in taxes.items()]
+        # Check if all taxes lines have account_id assigned. If not, there are repartition lines of the tax that have no account_id.
+        tax_names_no_account = [line['name'] for line in tax_vals if line['account_id'] == False]
+        if len(tax_names_no_account) > 0:
+            error_message = _(
+                'Unable to close and validate the session.\n'
+                'Please set corresponding tax account in each repartition line of the following taxes: \n%s'
+            ) % ', '.join(tax_names_no_account)
+            raise UserError(error_message)
+
+        MoveLine.create(
+            tax_vals
+            + [self._get_sale_vals(key, **amounts) for key, amounts in sales.items()]
+            + [self._get_stock_expense_vals(key, **amounts) for key, amounts in stock_expense.items()]
+            + [self._get_split_receivable_vals(key, **amounts) for key, amounts in split_receivables.items()]
+            + [self._get_combine_receivable_vals(key, **amounts) for key, amounts in combine_receivables.items()]
+        )
+
+        ## SECTION: Create cash statement lines and cash move lines
+        # Create the split and combine cash statement lines and account move lines.
+        # Keep the reference by statement for reconciliation.
+        # `split_cash_statement_lines` maps `statement` -> split cash statement lines
+        # `combine_cash_statement_lines` maps `statement` -> combine cash statement lines
+        # `split_cash_receivable_lines` maps `statement` -> split cash receivable lines
+        # `combine_cash_receivable_lines` maps `statement` -> combine cash receivable lines
+        statements_by_journal_id = {statement.journal_id.id: statement for statement in self.statement_ids}
+        # handle split cash payments
+        split_cash_statement_line_vals  = defaultdict(list)
+        split_cash_receivable_vals      = defaultdict(list)
```

Comment:
don't allign the "=" symbol like this, only one space (in all code) because it is not recommended in PEP8.

The problem with this kind of indentation is if you add a new variable in future which is longer than the longest existing one. You'll have to make a diff on all variables to allign them.